tests/buffer/buffer_constructors_common.h

```diff
@@ -200,278 +193,266 @@ class buffer_ctors {
       }
       std::vector<T> ref(cont1.size());
       std::iota(ref.begin(), ref.end(), 0);
-      if (!std::equal(ref.cbegin(), ref.cend(), cont1.cbegin(),
-                      value_operations::are_equal<T, T>) ||
-          !std::equal(ref.cbegin(), ref.cend(), cont2.cbegin(),
-                      value_operations::are_equal<T, T>) ||
-          !alloc_log.str().empty())
-        FAIL(log, "(Container, allocator) constructor fail.");
+      CHECK(std::equal(ref.cbegin(), ref.cend(), cont1.cbegin(),
+                       value_operations::are_equal<T, T>));
+      CHECK(std::equal(ref.cbegin(), ref.cend(), cont2.cbegin(),
+                       value_operations::are_equal<T, T>));
+      CHECK(alloc_log.str().empty());
     }
 
-    /* check (shared pointer, range) constructor*/
     {
-      std::shared_ptr<T> data(new T[size]);
-      std::fill(data.get(), (data.get() + size), 0);
+      INFO("check (shared pointer, range) constructor");
+      std::shared_ptr<T> data(new T[size], std::default_delete<T[]>());
```

Comment:
@kopylovanat By rereading your comment

> But on cppreference for constructor `template< class Y > shared_ptr( Y* ptr )` of `std::shared_ptr<T>` it is stated that "3) Uses the delete-expression `delete ptr` if `T` is not an array type; delete[] ptr if T is an array type (since C++17) as the deleter."
And in tested case T is not an array type, so it seems delete ptr will be used, not delete[] ptr. Am I missing something here?

and looking again at the C++17 draft
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4660.pdf
I think I missed the very important point of your comment: *in tested case T is not an array type*.
Yes, you are right, `new T[size]` does return a pointer `T*` and not an array `T[]`. :-(
OK, I think I understand now. I was thinking you were implementing the test from line 215 with `std::shared_ptr<T[]>`.
But actually this is the test really testing with a `std::shared_ptr<T>`!
So yes, you need to provide the real deleter corresponding to the new.